runtime/src/main/kotlin/kotlin/Throwable.kt

```diff
@@ -42,43 +43,150 @@ public open class Throwable(open val message: String?, open val cause: Throwable
+                val element = stackTraceStrings[frameIndex]
+                target.append(indent).append("    at ").append(element).endln()
+            }
+            if (commonFrames > 0) {
+                target.append(indent).append("    ... and ").append(commonFrames).append(" more common stack frames skipped").endln()
+            }
+            val suppressed = suppressedExceptionsList
+            if (!suppressed.isNullOrEmpty()) {
+                val suppressedIndent = indent + "    "
+                for (s in suppressed) {
+                    s.dumpFullTrace(suppressedIndent, "Suppressed: ")
+                }
+            }
+            return true
+        }
 
-        var cause = this.cause
-        while (cause != null) {
-            // TODO: should skip common stack frames
-            writeln("Caused by: $cause")
-            writeStackTraceElements(cause, writeln)
-            cause = cause.cause
+        private fun Throwable.commonStackFrames(): Int {
+            if (top === this) return 0
+            val topStack = top.stackTrace
+            val topSize = topStack.size
+            val thisStack = this.stackTrace
+            val thisSize = thisStack.size
+            val maxSize = minOf(topSize, thisSize)
+            var frame = 0
+            while (frame < maxSize) {
+                if (thisStack[thisSize - 1 - frame] != topStack[topSize - 1 - frame]) break
+                frame++
+            }
+            return frame
         }
     }
 
+
+    /**
+     * Returns the short description of this throwable consisting of
+     * the exception class name (fully qualified if possible)
+     * followed by the exception message if it is not null.
+     */
     public override fun toString(): String {
         val kClass = this::class
         val s = kClass.qualifiedName ?: kClass.simpleName ?: "Throwable"
         return if (message != null) s + ": " + message.toString() else s
     }
+
+    internal var suppressedExceptionsList: MutableList<Throwable>? = null
 }
 
 @SymbolName("Kotlin_getCurrentStackTrace")
 private external fun getCurrentStackTrace(): NativePtrArray
 
 @SymbolName("Kotlin_getStackTraceStrings")
 private external fun getStackTraceStrings(stackTrace: NativePtrArray): Array<String>
+
+/**
+ * Returns the detailed description of this throwable with its stack trace.
+ *
+ * The detailed description includes:
+ * - the short description (see [Throwable.toString]) of this throwable;
+ * - the complete stack trace;
+ * - detailed descriptions of the exceptions that were [suppressed][suppressedExceptions] in order to deliver this exception;
+ * - the detailed description of each throwable in the [Throwable.cause] chain.
+ */
+@SinceKotlin("1.4")
+public actual fun Throwable.stackTraceToString(): String = dumpStackTrace()
+
+/**
+ * Prints the [detailed description][Throwable.stackTraceToString] of this throwable to the standard error output.
```

Comment:
Currently it prints to the standard output, not the error output, so need to correct docs here.

I think it would be inconsistent being able to output to stderr exceptions only, while we don't expose general API for that in Kotlin.